holoviews/core/spaces.py

```diff
@@ -580,7 +601,10 @@ class DynamicMap(HoloMap):
        the cache is full.""")
 
     def __init__(self, callback, initial_items=None, **params):
-        if not isinstance(callback, Callable):
+
+        if isinstance(callback, types.GeneratorType):
+            callback = Generator(callback)
+        elif not isinstance(callback, Callable):
             callback = Callable(callback)
```

Comment:
Trivial conditional switch between ``Callable`` and ``Generator`` - a specific type of ``Callable``.